x-pack/plugins/enterprise_search/common/connectors/connectors.ts

```diff
@@ -129,16 +129,27 @@ export const CONNECTOR_DEFINITIONS: ConnectorServerSideDefinition[] = [
     serviceType: 's3',
   },
   {
-    iconPath: 'sharepoint_online.svg',
-    isBeta: false,
+    iconPath: 'sharepoint.svg',
+    isBeta: true,
     isNative: false,
-    isTechPreview: true,
+    isTechPreview: false,
     keywords: ['sharepoint', 'cloud', 'connector'],
     name: i18n.translate('xpack.enterpriseSearch.content.nativeConnectors.sharepoint.name', {
-      defaultMessage: 'Sharepoint Online & Server',
+      defaultMessage: 'Sharepoint Server',
     }),
     serviceType: 'sharepoint',
   },
+  {
+    iconPath: 'sharepoint_online.svg',
+    isBeta: false,
+    isNative: true,
+    isTechPreview: false,
+    keywords: ['sharepoint', 'office365', 'cloud', 'connector'],
+    name: i18n.translate('xpack.enterpriseSearch.content.nativeConnectors.sharepoint_online.name', {
+      defaultMessage: 'Sharepoint Online',
+    }),
+    serviceType: 'sharepoint_online',
```

Comment:
Afaik from talking to @danajuratoni this connector was in preview and we are not planning to have a compatible migration.

Older Sharepoint Connector will raise an error if it would be used for Sharepoint Online. Error will say something about "please migrate to Sharepoint Online connector).